src/compose/images.ts

```diff
@@ -417,35 +434,75 @@ async function getImagesForCleanup(): Promise<string[]> {
 		.value();
 }
 
-export async function inspectByName(
-	imageName: string,
-): Promise<Docker.ImageInspectInfo> {
-	try {
-		const image = await docker.getImage(imageName);
-		return await image.inspect();
-	} catch (e) {
-		if (NotFoundError(e)) {
-			const digest = imageName.split('@')[1];
-			let imagesFromDb: Image[];
-			if (digest != null) {
-				imagesFromDb = await db
-					.models('image')
-					.where('name', 'like', `%@${digest}`);
-			} else {
-				imagesFromDb = await db
-					.models('image')
-					.where({ name: imageName })
-					.select();
-			}
+export async function inspectByName(imageName: string) {
+	// Fail fast
+	assert(!!imageName, 'image name to inspect cannot be null');
```

Comment:
If `inspectByName` is passed a falsey value, this error message will be displayed in the journal logs right? Should we have a more user-friendly message? 

This error is also going to be of type AssertionError, what about changing it to `NotFoundError` which we already have, and which would enable more targeted handling of this class of errors? Or is there some complication associated with this error type change?

Reply:
If this function is passed a falsey value we should catch it before it getting to production because it's most likely a bug, this assertion is to catch those cases. For instance, the API tests I disabled were not failing before even though they were passing a null to this function. I'm ok with changing to a better message.

Reply:
yeah, we want it to throw if `imageName` is null though right ?

Reply:
I mean we won't get an assertion error, we will get something like 

```
Uncaught TypeError: Cannot read property 'length' of null
```

Reply:
oh yeah true. Well then I have no input besides considering the error message thrown that I suggested.

Reply:
I suppose we're removing this line after all the E2E tests pass?

Reply:
I'm wasn't sure what Felipe meant by "catch it before it getting to production". Reading Christina's comment now is confirming I'm definitely out of the loop. @pipex why is the assert added ? I thought it was to prevent any issues at runtime that the TS compiler isn't catching.

Reply:
@cywang117 no we won't remove them, this serves the same purpose as the `InternalConsistencyError` in other places of the code, which is to show bugs instead of hiding them. As @20k-ultra says, this is to catch issues at runtime that the compiler isn't catching.

See this also http://wiki.c2.com/?ShipWithAssertionsOn